### PR TITLE
Update creating service bindings

### PR DESCRIPTION
**Pull request by baijum**

### CODE REVIEW

File: docs/creating-service-bindings/creating-service-binding.md [hunk 1]
```diff
@@ -4,8 +4,67 @@ sidebar_position: 1
...
+5. Detect Bindings through owned resources
+
+If it's not the case, please refer to the "exposing binding data" section.
+
+You also required the application ready to create the ServiceBinding resource.
+
+There are two APIs supported by the Service Binding Operator.  Both has
+`ServiceBinding` as the same name for the kind.  But the API group is different.
+
+1. binding.operators.coreos.com - This API will be removed in future.
```
Comment (pedjak) [inline docs/creating-service-bindings/creating-service-binding.md]: Let's not talk about it in the userdoc, cc @slemeur

**Review by pedjak — COMMENTED** [review]
this section is missing various details about the options that are available to users (mappings, naming strategy, binding as env vars/files, etc.)

File: docs/creating-service-bindings/creating-service-binding.md (hunk 1 above)
Comment (baijum) [reply docs/creating-service-bindings/creating-service-binding.md]: Do you mean that we only need to document spec API?

Comment (pedjak) [reply docs/creating-service-bindings/creating-service-binding.md]: ah, sorry - misunderstanding - I would not mention here at all any deprecation plans. We should describe both apis and give where needed details what is different.

Comment (baijum) [reply docs/creating-service-bindings/creating-service-binding.md]: Done. dc17740535825e50ffe82736d6b9226ac6bcb631

File: docs/creating-service-bindings/creating-service-binding.md [hunk 2]
```diff
@@ -4,8 +4,68 @@ sidebar_position: 1
 
 # Creating Service Binding
 
-In this section, we consider that the service you want to connect to is already exposing binding metadata by either:
-- Beeing part of the "known" services
-- Exposing binding-metadatas by default
-If it's not the case, please refer to the following section(#)
+In this section, we consider that the service you want to connect to is already
```
Comment (slemeur) [inline docs/creating-service-bindings/creating-service-binding.md:7]: We should provide a link to the other section of the documentation which explains how to do that.

File: docs/creating-service-bindings/creating-service-binding.md [hunk 3]
```diff
@@ -4,8 +4,68 @@ sidebar_position: 1
...
+You also required the application ready before creating the ServiceBinding
+resource.
+
+There are two APIs supported by the Service Binding Operator.  Both has
+`ServiceBinding` as the same name for the kind.  But the API group is different.
+
+1. binding.operators.coreos.com
+2. service.binding (API group used in the [Service Binding Spec][spec])
+
+Here is the example configuration with `binding.operators.coreos.com` API group:
```
Comment (slemeur) [inline docs/creating-service-bindings/creating-service-binding.md:25]: we should be providing more guidance.
We should provide a step by step guide for that:
- What do you need to create the service binding
- When to create it? Does the service and application needs to be already setup. What if they are not yet available or up?
- What are the different outputs it should be possible to get?
- Should I setup the SB via kubectl or? 
- I think we should have an explaination here about how to do it on OpenShift Developer Console

File: docs/creating-service-bindings/creating-service-binding.md [hunk 4]
```diff
@@ -4,8 +4,68 @@ sidebar_position: 1
...
+4. Secret Generated through OLM Descriptors
+5. Detect Bindings through owned resources
+
+You also required the application ready before creating the ServiceBinding
+resource.
+
+There are two APIs supported by the Service Binding Operator.  Both has
+`ServiceBinding` as the same name for the kind.  But the API group is different.
+
+1. binding.operators.coreos.com
```
Comment (slemeur) [inline docs/creating-service-bindings/creating-service-binding.md]: Which one should I use?
We should explain why we have the two apis and the impact?

File: docs/creating-service-bindings/binding-options.md [hunk 5]
```diff
@@ -6,9 +6,14 @@ sidebar_position: 2
```
Comment (slemeur) [inline docs/creating-service-bindings/binding-options.md:6]: We should have an option that describe/list all the potential options and when a user would be interested by those

File: docs/creating-service-bindings/binding-options.md [hunk 6]
```diff
@@ -6,9 +6,14 @@ sidebar_position: 2
 
 ## Transform binding names before projection into application
 
-Binding names declared through annotations or CSV descriptors are processed before injected into the application according to the following strategy
- - names are upper-cased
- - service resource kind is upper-cased and prepended to the name
+This feature is only available for `ServiceBinding` in the
```
Comment (slemeur) [inline docs/creating-service-bindings/binding-options.md]: maybe we could use a warning visual?  not sure how to do that with docusaurus :)

File: docs/creating-service-bindings/binding-options.md [hunk 7]
```diff
@@ -6,9 +6,14 @@ sidebar_position: 2
 
 ## Transform binding names before projection into application
```
Comment (slemeur) [inline docs/creating-service-bindings/binding-options.md:7]: We should start by the use case and when a user would be interested by that

File: docs/creating-service-bindings/binding-options.md [hunk 8]
```diff
@@ -6,9 +6,14 @@ sidebar_position: 2
...
+`binding.operators.coreos.com` API group.
+
+Binding names declared through annotations or OLM descriptors are processed
+before injected into the application according to the following strategy
+
+- names are upper-cased
+- service resource kind is upper-cased and prepended to the name
 
 example:
 ```yaml
```
Comment (slemeur) [inline docs/creating-service-bindings/binding-options.md:28]: I'm not following the sample.

File: docs/creating-service-bindings/binding-options.md [hunk 9]
```diff
@@ -97,13 +104,22 @@ port: 8080
 3. `title` - Title case all letters.
 
 
-## Compose custom binding variables <kbd>TBU</kbd>
+## Compose custom binding variables
 
-If the backing service doesn't expose binding metadata or the values exposed are not easily consumable, then an application author may compose custom binding variables using attributes in the Kubernetes resource representing the backing service.
+This feature is only available for `ServiceBinding` in the
```
Comment (slemeur) [inline docs/creating-service-bindings/binding-options.md:113]: we should have a visual warning here

File: docs/creating-service-bindings/binding-options.md [hunk 10]
```diff
@@ -97,13 +104,22 @@ port: 8080
...
+variables using attributes in the Kubernetes resource representing the backing
+service.
 
-Example, the backing service CR may expose the host, port and database user in separate variables, but the application may need to consume this information as a connection string.
+The *custom binding variables* feature enables application authors to request
+customized binding secrets using a combination of Go and jsonpath templating.
+
+Example, the backing service CR may expose the host, port and database user in
+separate variables, but the application may need to consume this information as
+a connection string.
```
Comment (slemeur) [inline docs/creating-service-bindings/binding-options.md:126]: like this sample use case :)

File: docs/creating-service-bindings/custom-path-injection.md [hunk 11]
```diff
@@ -6,9 +6,12 @@ sidebar_position: 3
```
Comment (slemeur) [inline docs/creating-service-bindings/custom-path-injection.md:6]: We should have an intro that explain what that is about, the different options, when to use them.

**Review by slemeur — CHANGES_REQUESTED** [review]

**Pull request merged**